app/services/user_event_creator.rb

```diff
@@ -79,18 +79,11 @@ def build_disavowal_token
   # @return [Array(Event, String)] an (event, disavowal_token) tuple
   def create_event_for_new_device(event_type:, user:, disavowal_token:)
     if user.fully_registered? && user.has_devices? && disavowal_token.nil?
-      device, event, disavowal_token = Device.transaction do
+      event, disavowal_token = Device.transaction do
```

Comment:
Need to investigate if there was a reason these properties weren't just pulled out of the `event` model, e.g. if it isn't automatically cached and incurs an additional database query.

Reply:
> Still need to investigate whether it's fine to access `device` via `event`, but since we'll eventually be removing all of the code which accesses `event.device` in `AlertUserAboutNewDevice` (behind feature flag), maybe it's a moot point.

Based on current build failures, I'm going to assume there was some optimization intended here, so I'll restore that and we can deal with removing the `device` parameter once we get to the point of removing the feature flag.

Example failure:

```
     Failure/Error: @app.call(env)
     Bullet::Notification::UnoptimizedQueryError:
       user: root
       POST /login/two_factor/personal_key
       USE eager loading detected
         Event => [:device]
         Add to your query: .includes([:device])
```